pkg/querysharding/analyzer.go

```diff
@@ -55,15 +57,14 @@ func (a *QueryAnalyzer) Analyze(query string) (QueryAnalysis, error) {
 			}
 		case *parser.BinaryExpr:
 			if n.VectorMatching != nil {
-				analysis = analysis.scopeToLabels(n.VectorMatching.MatchingLabels, n.VectorMatching.On)
+				shardingLabels = without(n.VectorMatching.MatchingLabels, []string{"le"})
+				analysis = analysis.scopeToLabels(shardingLabels, n.VectorMatching.On)
```

Comment:
Same here, let's try to avoid the global variable.

```suggestion
				shardingLabels =: without(n.VectorMatching.MatchingLabels, []string{"le"}) 
				analysis = analysis.scopeToLabels(shardingLabels, n.VectorMatching.On)
```